mreg/api/v1/serializers.py

```diff
@@ -85,19 +85,21 @@ class Meta:
 
     def validate(self, data):
         """
-        Make sure a macaddress is semi-unique:
+        Make sure a mac address is semi-unique:
         - Unique if the IP is not in a network.
         - Only in use by one IP per network.
-        - If the network has a vlan id, make sure it is only in use by one of
-          the networks on the same vlan. Exception: allow both a ipv4 and ipv6
-          address on the same vlan to share the same mac address.
         """
 
         def _raise_if_mac_found(qs, mac):
-            if qs.filter(macaddress=mac).exists():
-                inuse_ip = qs.get(macaddress=mac).ipaddress
-                raise serializers.ValidationError(
-                    "macaddress already in use by {}".format(inuse_ip))
+            # There is a theoretical possibility that a mac address can be in use
+            # by multiple IP addresses, although normally it would never be more than 1.
+            inuse_set = qs.filter(macaddress=mac)
+            if inuse_set.exists():
+                ips = []
+                for inuse_ip in inuse_set:
+                    ips.append("{}".format(inuse_ip.ipaddress))
```

Comment:
Better:
```
ips = inuse_set.values_list('ipaddress', flat=True)
```

Reply:
Made a commit just now with your suggested change.

As for the vlan restriction, there's a test for it that looks correct to me:
https://github.com/unioslo/mreg/blob/7af8098a4a468733e3bee2516bc51ab82a613395/mreg/api/v1/tests/tests.py#L1288-L1310

I also tested it with mreg-cli just now. Ended up with this:
```
mreg> host info eple
Name:         eple.uio.no
Contact:
A_Records     IP              MAC
              10.0.0.5        aa:bb:cc:dd:ee:ff
TTL:          (Default)
mreg> host info banan
Name:         banan.uio.no
Contact:
AAAA_Records  IP              MAC
              2001:db8:1::4   aa:bb:cc:dd:ee:ff
TTL:          (Default)
mreg> network info 10.0.0.0/24
Network:                 10.0.0.0/24
Netmask:                 255.255.255.0
Description:             test1
Category:
Location:
VLAN                     1
DNS delegated:           False
Frozen                   False
IP-range:                10.0.0.0 - 10.0.0.255
Reserved host addresses: 3
                         10.0.0.0 (net)
                         10.0.0.1
                         10.0.0.2
                         10.0.0.3
                         10.0.0.255 (broadcast)
Used addresses:          1
Unused addresses:        250 (excluding reserved adr.)
mreg> network info 2001:db8:1::/64
Network:                 2001:db8:1::/64
Netmask:                 ffff:ffff:ffff:ffff:0000:0000:0000:0000
Description:             hei og hå
Category:
Location:
VLAN                     1
DNS delegated:           False
Frozen                   False
IP-range:                2001:db8:1:: - 2001:db8:1:0:ffff:ffff:ffff:ffff
Reserved host addresses: 3
                         2001:db8:1:: (net)
                         2001:db8:1::1
                         2001:db8:1::2
                         2001:db8:1::3
Used addresses:          1
Unused addresses:        18446744073709551611 (excluding reserved adr.)
mreg>
```